### ISSUE TITLE
==/!= doesn't work on optional type

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#1824]
Behold:
```python
def f(a: Optional[str], b: Optional[str]) -> bool:
  return a == b
```
We get this beautiful error with `--strict-optional`:
```
error: Unsupported left operand type for == (some union)
```

**Comment by rwbarton** [comment]
Maybe not obvious, but #1766 was a reduction of this issue.

**Comment by ddfisher** [comment]
No longer gives an error.  I think this was fixed along with #1766 by #1836.
